nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -403,11 +423,215 @@ async fn run() -> Result<()> {
+        Ok(None) => {
+            None
+        }
+        Err(err) => {
+            tracing::debug!("Failed to parse identifier, error: {}", err);
+            None
+        }
+    }
+}
+
+async fn reward(
+    assign_event_id: String,
+    workers: &Vec<Sender>,
+    payload: ResultPayload,
+    classifier_result_collection: &Collection<ClassifierResult>,
+    collection: &Collection<FinishedJobs>,
+    job_type: u16,
+) -> bool {
+    tracing::debug!(
+        "All the results for {} are consistent, sending",
+        assign_event_id
+    );
+    let raw_data_id = payload.header.raw_data_id.clone();
+    let result_identifier = get_result_identifier(payload.output.clone());
+    match result_identifier   {
+        Err(_) =>{},
+        Ok(None)=>{},
+        Ok(Some(tag_id)) =>{
+            // Save classifier result to db
+            let classifier_result = ClassifierResult {
+                _id: raw_data_id,
+                kv_key: payload.kv_key.clone(),
+                tag_id
+            };
+            match classifier_result_collection.insert_one(classifier_result, None).await {
+                Ok(_) => {
+                    tracing::info!("Write classifier result to db: {}", payload.kv_key.clone());
+                }
+                Err(err) => {
+                    tracing::error!("Cannot write classifier result to db: {}", err);
+                }
+            }
+        }
+    }
+
+    let db_entry = FinishedJobs {
+        _id: raw_data_id,
+        workers: workers.clone(), // if we remove first, no need to clone but this is safer
+        timestamp: chrono::Utc::now().timestamp() as u64,
+        result: payload,
+        job_type,
+        assign_event_id,
+    };
+    tracing::debug!(
+        "Rewarding all workers for data: {}",
+        raw_data_id.to_string()
+    );
+    match collection.insert_one(db_entry, None).await {
+        Ok(_) => {
+            FINISHED_JOBS.inc();
+            true
+        }
+        Err(err) => {
+            // We log and that's it, we keep the entry in book for later inspection
+            tracing::error!("Cannot write finished job to db: {}", err);
+            FAILED_JOBS.inc();
+            false
+        }
+    }
+
+}
+
 fn get_result_payload(ev: &RelayEvent) -> Result<ResultPayload> {
+    tracing::debug!("Content is: {}", (ev.event.content.as_str()));
     let res: ResultPayload = serde_json::from_str(ev.event.content.as_str())?;
     Ok(res)
 }
 
+ /**
+     * For now, only use the first item in array to compare results
+     * result example: [{"tag_id":2867,"distance":0.5045340279460676}]
+     * return example: 2867
+*/
+pub fn get_result_identifier(output:String) -> Result<Option<String>> {
+    tracing::debug!("output: {}", output);
+    let result_arr: Vec<ClassifierJobOutput> =
+        serde_json::from_str(output.as_str())?;
+    tracing::debug!("result_arr: {:?}", result_arr);
+    let answer = result_arr.get(0);
+    tracing::debug!("answer is : {:?}", answer);
```

Comment:
It seems redundant, given we already log below both when it's None and when it's Some..

Also, because of this log we are not writing it as

```rust
match result.get(0) {
    ...
}
```